controller/src/main/java/org/jboss/as/controller/persistence/yaml/YamlConfigurationExtension.java

```diff
@@ -524,6 +549,25 @@ public String getCommandLineInstructions() {
         return MGMT_OP_LOGGER.argYaml();
     }
 
+    @SuppressWarnings("unchecked")
+    private void processUnmanagedDeployments(ImmutableManagementResourceRegistration rootRegistration, Map.Entry<String, Object> deployment, Map<PathAddress, ParsedBootOp> xmlOperations, List<ParsedBootOp> postExtensionOps) {
+        String name = deployment.getKey();
+        OperationEntry operationEntry = rootRegistration.getOperationEntry(PathAddress.pathAddress("deployment", name), ADD);
+        if (deployment.getValue() != null && deployment.getValue() instanceof Map) {
+            Map<String, Object> attributes = (Map<String, Object>) deployment.getValue();
+            Map<String, Object> content = (Map<String, Object>) (((Iterable<? extends Object>) attributes.get("content")).iterator().next());
+            Set<String> result = content.keySet().stream().distinct().filter(MANAGED_CONTENT_ATTRIBUTES::contains).collect(Collectors.toSet());
+            if (!result.isEmpty()) {
+                throw MGMT_OP_LOGGER.unsupportedDeployment(name, result);
+            }
+            if(attributes.containsKey("enabled")) {
+                boolean enabled = (Boolean) attributes.getOrDefault("enabled", "false");
```

Comment:
Is L563-565 cruft? This seems to be validation logic and I don't see why 'enabled' needs special handling.

L564 is confusing, as String "false" will result in ClassCastException when cast to Boolean.

L563 has already established that getOrDefault will never use the default value though, so it doesn't break anything, it's just fragile.

Cleaner is replacing the entire L563-565 block with

// Confirm we have a valid boolean 'enabled' setting, if present
(Boolean) attributes.getOrDefault("enabled", Boolean.FALSE);

But.... why is that needed, and if validation is needed a validation check that uses CCE to signal failure isn't very user friendly.

Reply:
Yes correct, those lies are cruft.